packages/mask/src/extension/popups/hooks/useFriendsFromSearch.tsx

```diff
@@ -1,43 +1,82 @@
-import { ECKeyIdentifier, EMPTY_LIST, type NextIDPersonaBindings } from '@masknet/shared-base'
+import {
+    ECKeyIdentifier,
+    EMPTY_LIST,
+    type NextIDPersonaBindings,
+    type EnhanceableSite,
+    type ProfileIdentifier,
+} from '@masknet/shared-base'
 import { uniqBy } from 'lodash-es'
 import { useMemo } from 'react'
 import type { Friend } from './useFriends.js'
 import { useCurrentLinkedPersona } from '@masknet/shared'
 import { profilesFilter } from './useFriendProfiles.js'
-import { PlatformSort } from '../pages/Friends/common.js'
+import { PlatformSort, type Profile } from '../pages/Friends/common.js'
 
-export type NextIDPersonaBindingsWithIdentifier = NextIDPersonaBindings & { linkedPersona: ECKeyIdentifier } & {
+export type NextIDPersonaBindingsWithIdentifier = Omit<NextIDPersonaBindings, 'proofs'> & { proofs: Profile[] } & {
+    linkedPersona: ECKeyIdentifier
+} & {
     isLocal?: boolean
 }
 
 export function useFriendsFromSearch(
+    localSearchedResult?: Friend[],
     searchResult?: NextIDPersonaBindings[],
     localList?: Friend[],
     searchValue?: string,
 ): NextIDPersonaBindingsWithIdentifier[] {
     const currentIdentifier = useCurrentLinkedPersona()
     return useMemo(() => {
-        if (!searchResult?.length) return EMPTY_LIST
+        if (!searchResult?.length && !localSearchedResult?.length) return EMPTY_LIST
+        const localProfiles: NextIDPersonaBindingsWithIdentifier[] =
+            localSearchedResult
+                ?.filter((x) => x.persona.publicKeyAsHex !== currentIdentifier?.identifier.publicKeyAsHex)
+                .map((item) => {
+                    const profile = item.profile as ProfileIdentifier
+                    return {
+                        proofs: [
+                            {
+                                platform: profile?.network as
+                                    | EnhanceableSite.Twitter
+                                    | EnhanceableSite.Facebook
+                                    | EnhanceableSite.Instagram,
+                                identity: profile?.userId,
+                                is_valid: true,
+                                last_checked_at: '',
+                                name: profile?.userId,
+                                created_at: '',
+                            },
```

Comment:
Make it cleaner, this change would be also beneficial for TS performance.
You can hoist `FriendNetwork` to upper scope.

```suggestion
                .map((item) => {
                    type FriendNetwork = EnhanceableSite.Twitter
                                    | EnhanceableSite.Facebook
                                    | EnhanceableSite.Instagram
                    const profile = item.profile as ProfileIdentifier
                    return {
                        proofs: [
                            {
                                platform: profile.network as FriendNetwork,
                                identity: profile.userId,
                                is_valid: true,
                                last_checked_at: '',
                                name: profile.userId,
                                created_at: '',
                            },
```